src/trace.rs

```diff
@@ -0,0 +1,309 @@
+        // normalized tracing metadata for that log record.
+        let norm_meta = event.normalized_metadata();
+        let meta = norm_meta.as_ref().unwrap_or_else(|| event.metadata());
+
+        let level = match meta.level() {
+            &Level::TRACE => "TRCE",
+            &Level::DEBUG => "DBUG",
+            &Level::INFO => "INFO",
+            &Level::WARN => "WARN",
+            &Level::ERROR => "ERR!",
+        };
+        let uptime = clock::now() - self.start_time;
+        write!(
+            f,
+            "{} [{:>6}.{:06}s] {}{} ",
+            level,
+            uptime.as_secs(),
+            uptime.subsec_micros(),
+            SpanContext(&span_ctx),
+            meta.target()
+        )?;
+        {
+            let mut recorder = span_ctx.new_visitor(f, true);
+            event.record(&mut recorder);
+        }
+        writeln!(f)
+    }
+}
+
+impl LevelHandle {
+    /// Returns a new `LevelHandle` without a corresponding filter.
+    ///
+    /// This will do nothing, but is required for admin endpoint tests which
+    /// do not exercise the `proxy-log-level` endpoint.
+    pub fn dangling() -> Self {
+        let builder = subscriber_builder()
+            .with_env_filter(EnvFilter::default())
+            .with_filter_reloading();
+        let inner = builder.reload_handle();
+        LevelHandle { inner }
+    }
+
+    pub fn set_level(&self, level: impl AsRef<str>) -> Result<(), Error> {
+        let level = level.as_ref();
+        let filter = level.parse::<EnvFilter>()?;
+        self.inner.reload(filter)?;
+        info!(message = "set new log level", %level);
+        Ok(())
+    }
+
+    pub fn current(&self) -> Result<String, Error> {
+        self.inner
+            .with_current(|f| format!("{}", f))
+            .map_err(Into::into)
+    }
+}
+
+impl fmt::Debug for LevelHandle {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.inner
+            .with_current(|c| {
+                f.debug_struct("LevelHandle")
+                    .field("current", &format_args!("{}", c))
+                    .finish()
+            })
+            .unwrap_or_else(|e| {
+                f.debug_struct("LevelHandle")
+                    .field("current", &format_args!("{}", e))
+                    .finish()
+            })
+    }
+}
+
+/// Implements `fmt::Display` for a `tokio-trace-fmt` span context.
+struct SpanContext<'a, N>(&'a Context<'a, N>);
+
+impl<'a, N> fmt::Display for SpanContext<'a, N> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut seen = false;
+        self.0.visit_spans(|_, span| {
+            write!(f, "{}", span.name())?;
+            seen = true;
+
+            let fields = span.fields();
+            if !fields.is_empty() {
+                write!(f, "{{{}}}", fields)?;
+            }
+            ":".fmt(f)
+        })?;
+        if seen {
+            f.pad(" ")?;
+        }
+        Ok(())
+    }
+}
```

Comment:
if we switch to the default ormatter as I mentioned above, this can all go away.